create

```diff
@@ -61,6 +61,13 @@ else
 fi
 
 mke2fs -Fjqt $OSP_FILESYSTEM $filesystem_dev
+
+# in some cases, mkfs will remap the partition tables, for example
+# /dev/mapper/foo-1 will turn into /dev/mapper/foo-part1. Rebuild the
+# map so we correct that:
+unmap_disk0 $blockdev
+filesystem_dev=$(map_disk0 $blockdev)
+
```

Comment:
Thanks for your finding. Do you use `multipath`? If `no_partitions` is not set in `multipath.conf(5)` it may be the cause for some asynchronous change in the partition mappings. If that does not help, it maybe an option to try `sfdisk` with the `--no-tell-kernel` option, because later `kpartx` will do the job in the create script.

If that doesn't work either I suggest to wrap your changes into a conditional and not to use `unmap_disk0`, because this will not remove `foo-part1`, just `foo-1`:
```bash
if [ ! -b "$filesystem_dev" ]; then
  if [ -b "$blockdev"-part1 ]; then
    kpartx -d -p-part $blockdev
  fi
  filesystem_dev=$(map_disk0 $blockdev)
fi
```

Reply:
wow, thanks! lots to think about, will try those out soon, thanks!

Reply:
@saschalucas I tested your proposed solution, and it works in our environment. However I'm wondering if a more proper fix would be to simply use `-p-part` instead of `-p-` when invoking `kpartx`, as this seems like the default partition delimiter used by the kernel itself?